test/Feature/EmbedDisambiguationSpec.hs

```diff
@@ -431,3 +431,9 @@ spec =
           [json|[{"id":1,"users":[{"id":1},{"id":3}]},{"id":2,"users":[{"id":1}]},{"id":3,"users":[{"id":1}]},{"id":4,"users":[{"id":1}]},{"id":5,"users":[{"id":2},{"id":3}]},{"id":6,"users":[{"id":2}]},{"id":7,"users":[{"id":2}]},{"id":8,"users":[]}]|]
           { matchHeaders = [matchContentTypeJson] }
 
+    context "m2m embed when the junction table is in an internal schema" $ do
+      -- https://github.com/PostgREST/postgrest/issues/1736
+      it "works by disregarding the internal junction" $
```

Comment:
I think we need another test-case for an internal junction with a different name (that has no view with the same name) - this should not be embeddable at all.

Reply:
Here you refer to this case you put up [here](https://github.com/PostgREST/postgrest/issues/1587#issuecomment-734995669) right?

> So just create two separate tables, both with the same fks and the same name, but obviously only one of them in the exposed schema. The error will be the same "could not find any relationship". Drop one of the tables and it works again.

Reply:
No, but that's a very good one to add, too!

The current test is just testing whether the view in the exposed schema can be embedded without an ambiguous error. You solved the bug, that this was not the case by removing the private junction completely. But you could have also solved it by giving "priority" to the public junction - and still allowing the private junction to be embedded somehow (the easiest case would be "only when no public junction with the same name exists"). I think this was a (bad) suggestion I made in the discussion in that issue.

To be 100% we're not breaking this again, you need to add a test:
```
TABLE public.A
TABLE public.B
JUNCTION private.C (A, B)
```
without any direct or indirect (through a view) exposure of `C`. This should fail with "No Relationship found".

Reply:
Now it's clear as day. Done!